### PR TITLE
Cult balance of the week #2

### LINKED ISSUE #20707: Cult 3 forge has a fucked up icon and description
#### Description of issue
image

#### Steps to reproduce
- Be cultist
- Summon forge
- Examine it
- Pic related

### LINKED ISSUE #20876: Various known cult bugs
That I'm putting here so I don't forget to fix them later.
* Apparently the detective scanner cannot find blood in runes
* Cult_Legacy structures (pylons, old forges, etc) get erased by Nar-Sie
* Soul Blades planted in altars produce shades that can't move or see their own blade
* Soul Blades can cast perforate from INSIDE an altar apparently, which causes bugs and stuff
* When you attack blood stones with a weapon, the flavor text is missing the attack verb
* Shades get stuck in doors
* Constructs should be affected by holy water somehow
* Constructs should be able to use some runes, namely those that don't require blood like Path runes.

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #20876
Fixes #20707

:cl:
* bugfix: Fixed cult forges having a borked (or rather missing) description.
* bugfix: Voice of Narsie admin communications now get properly logged again, and can be read by observers.
* bugfix: Fixed runes not properly carrying the DNA of the blood used to draw them, and detective scanners being unable to scan them.
* bugfix: Soul blades can no longer perforate from inside an altar.
* rscadd: Cultists now risk having centcom reveal their presence prematurely if over 40% of the players are cultists before the sacrifice has occurred.
* rscadd: Emitter beams now actually damage cult structures. They deal very little damage however, taking a few minutes to break down a blood stone. They offer themselves as a slow but safer way to handle them.
* rscadd: All constructs can now activate runes, most of them at least. Runes with a blood cost will require an external blood source. Summon Robes doesn't do anything when used by constructs.
* rscadd: Cult monkeys now have access to the cult spells, allowing them to write runes (though they need an external blood source, since monkeys don't have proper blood vessels). They also got back their custom monkey-sized cult robes from the Summon Robes rune.
* rscadd: Holy water can now damages constructs, and slows them down.
* tweak: Cult floors, walls and girders can no longer be clockworkified.
* tweak: Shades can no longer get stuck inside airlocks.
* imageadd: Cultifying xenoarch artifacts have new visual effects when they convert their surroundings.

### CODE REVIEW

**Comment by Irradiation** [comment]
aaaa

**Comment by DrSnips** [comment]
While it might never come up gyrotrons (the R-ust emitter things) should have some effect like the emitter change.

**Comment by DeityLink** [comment]
@DrSnips They fire an actual beam projectile, therefore they can already damage them.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -1043,6 +1043,16 @@
 	if(volume >= 5)
 		T.bless()
 
+/datum/reagent/holywater/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
+	..()
+
+	if(volume >= 5)
+		if(istype(M,/mob/living/simple_animal/construct))
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Reagents.dm:1050]: shouldn't be if (M.supernatural))?

That way, it affects things like zombies, vampires, etc.

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -1289,7 +1289,7 @@ About the new airlock wires panel:
 	return ..()
 
 /obj/machinery/door/airlock/Uncross(atom/movable/mover)
-	if(density && ismob(mover) && !(mover.checkpass(PASSGLASS) && !opacity) && !(mover.checkpass(PASSDOOR)))
+	if(density && ismob(mover) && !(mover.checkpass(PASSGLASS) && !opacity) && !(mover.checkpass(PASSDOOR)) && !(istype(mover,/mob/living/simple_animal/shade)))//REEEEEEE
```
Comment (MadmanMartian) [inline code/game/machinery/doors/airlock.dm:1292]: pls no hardcoderino

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 3]
```diff
@@ -1028,7 +1028,11 @@ var/list/datum/dna/hivemind_bank = list()
 	if(amount == 0)
 		return
 
-	var/mob/living/carbon/target = M.changeling_sting(amount, /obj/item/verbs/changeling/proc/changeling_chemsting, allow_self = bool=="Yes"?TRUE:FALSE)
+	var/mob/living/carbon/target
```
Comment (MadmanMartian) [inline code/game/gamemodes/changeling/changeling_powers.dm:1031]: what in tarnation is going on here?

File: code/datums/helper_datums/command_alerts.dm [hunk 4]
```diff
@@ -401,17 +410,17 @@ The access requirements on the Asteroid Shuttles' consoles have now been revoked
...
 	force_report = 1
 
 /datum/command_alert/bloodstones_anchor/announce()
 	message = "Occult energies from [station_name()] are reaching a critical point. A breach through space has materialized on one of the bloodstones. Destroy it at all costs, do not let any cultist near it."
 	..()
 
 /datum/command_alert/bloodstones_broken
-	name = "Occult Activity Ceased - Lock Down Removed"
-	alert_title = "Occult Assault"
+	name = "Occult Activity Ceased - Lock Down Lifed"
```
Comment (MadmanMartian) [inline code/datums/helper_datums/command_alerts.dm]: Lock down Lifted.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/modules/reagents/Chemistry-Reagents.dm:1050]: Could also be on the `handle_reagent` of the cultist role and check if the role reside withing a construct maybe

File: code/game/gamemodes/changeling/changeling_powers.dm (hunk 3 above)
Comment (ShiftyRail) [reply code/game/gamemodes/changeling/changeling_powers.dm:1031]: Most likely forgot to rebase

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (DeityLink) [reply code/modules/reagents/Chemistry-Reagents.dm:1050]: only constructs have the purge variable. and its meant to affect all constructs, not just cult ones, but only constructs. Other mobs might get their own code later.

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (DeityLink) [reply code/game/machinery/doors/airlock.dm:1292]: It's literally the only mob in the entire game with that property though. so either I had a snowflake exception here, or I had a snowflake flag or variable that'll only be used here as well. Which should it be?

File: code/game/gamemodes/changeling/changeling_powers.dm (hunk 3 above)
Comment (DeityLink) [reply code/game/gamemodes/changeling/changeling_powers.dm:1031]: My code just won't compile unless I break up that ternary operator. Beats me.

Comment (Exxion) [reply code/game/gamemodes/changeling/changeling_powers.dm:1031]: Wrong button
Also that ternary operator is very stupid
See if you can spot why

File: code/datums/helper_datums/command_alerts.dm (hunk 4 above)
Comment (DeityLink) [reply code/datums/helper_datums/command_alerts.dm]: fix'd

File: code/game/gamemodes/changeling/changeling_powers.dm (hunk 3 above)
Comment (Intigracy) [reply code/game/gamemodes/changeling/changeling_powers.dm:1031]: jesus that ternary

**Pull request merged**